src/annotator/test/guest-test.js

```diff
@@ -721,6 +721,72 @@ describe('Guest', () => {
         });
       });
     });
+
+    describe('on "renderThumbnail" event', () => {
+      const makeCallback = () => {
+        const { promise, resolve, reject } = Promise.withResolvers();
```

Comment:
Is `Promise.withResolvers()` polyfilled by TypeScript yet? If so, maybe it's time to get rid of our own `promiseWithResolvers()` function.

Reply:
TypeScript doesn't add polyfills. I think you meant Babel? In test code we can use `Promise.withResolvers` because we only run tests in more recent browsers. In production code we generally use [ponyfills](https://github.com/sindresorhus/ponyfill) rather than polyfills for APIs that are not yet widely supported enough.

Reply:
> TypeScript doesn't add polyfills.

Oh, really? I thought they did add some polyfills based on your tsconfig's `target` 😅 

> In test code we can use Promise.withResolvers because we only run tests in more recent browsers.

Right! I didn't realize this was a test.